### PR TITLE
Fix regression in scroll zoom behavior

### LINKED ISSUE #5985: Scroll zoom performance regresssion in v0.43.0
**mapbox-gl-js version**:

### Steps to Trigger Behavior

Use mouse scroll to zoom without pausing for about 2 second in 0.43.0
Do the same in version 0.42.2

### Expected Behavior

Smooth zooming

### Actual Behavior

The zooming is not smooth since version 0.43.0

### PR DESCRIPTION
**Pull request by anandthakker**
Closes #5985

### CODE REVIEW

**Comment by anandthakker** [comment]
Alright, thanks to help from @kkaefer and @asheemmamoowala, I think this _should_ be ready to go once it gets a 👍 review.

File: src/ui/handler/scroll_zoom.js [hunk 1]
```diff
@@ -4,9 +4,23 @@ const DOM = require('../../util/dom');
...
 import type Map from '../map';
 import type Point from '@mapbox/point-geometry';
+import type Transform from '../../geo/transform';
+
+// deltaY value for mouse scroll wheel identification
+const wheelZoomDelta = 4.000244140625;
+// These magic numbers control the rate of zoom. Trackpad events fire at a greater
+// frequency than mouse scroll wheel, so reduce the zoom rate per wheel tick
+const defaultZoomRate = 1 / 100;
+const wheelZoomRate = 1 / 500;
```
Comment (kkaefer) [inline src/ui/handler/scroll_zoom.js]: I think I prefer the faster scrolling of `400` here, but `500` works well too.

Comment (anandthakker) [reply src/ui/handler/scroll_zoom.js]: I split the difference 😄

**Comment by anandthakker** [comment]
Updated top with summary of the problem & changes

**Comment by hyperknot** [comment]
I _think_ this one introduced the occassional "Failed to invert matrix" error, which breaks the map. I've recommended a simple fix there.
https://github.com/mapbox/mapbox-gl-js/issues/6486

**Pull request merged**